src/net.rs

```diff
@@ -177,6 +178,44 @@ pub trait HostResponse {
     fn finish(&mut self) -> Result<(), Error>;
 }
 
+/// Represents a physical port that can be used to interact with client devices.
+///
+/// This for example is used for a PA RoT to send messages to a AC RoT.
+///
+/// This trait provides a generic mechanism for sending and receiving requests
+/// from devices.
+pub trait DevicePort<'a> {
+    /// The send function takes a `header` and `msg` to send to a device.
+    /// The implementation of this function must send the header and msg
+    /// to the device specified by `dest`.
+    ///
+    /// This can be either a blocking of non-blocking call. On success it
+    /// should return the number of bytes sent. If the number of bytes sent
+    /// is 0, but not an error, then the operation is asynchronous. In this
+    /// case there should be an out of band method to indicate the send is
+    /// complete.
```

Comment:
A few thoughts:
- The return value should definitely be an enum, rather than relying on a sentinel.
- What would an "out-of-band" mechanism look like? The problem is that this is probably going to be in the guts of a Manticore state machine, which isn't exactly designed for asynchronicity. How do you envision this trait being used?

Reply:
I have improved this, I'm going to mark this as resolved as it has been removed.